src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,49 +1,115 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
+import classNames from 'classnames';
 import { Loader } from '../Loader';
+import { LoadingError } from '../LoaderError/LoaderError';
+import { Todo } from '../../types/Todo';
+import { User } from '../../types/User';
+import { getUser } from '../../api';
+
+type Props = {
+  todo: Todo;
+  onClose: () => void;
+};
+
+export const TodoModal: React.FC<Props> = ({
+  todo,
+  onClose,
+}) => {
+  const {
+    id,
+    title,
+    completed,
+    userId,
+  } = todo;
+
+  const [user, setUser] = useState<User>();
+  const [isLoading, setIsLoading] = useState(false);
+  const [hasLoadingError, setHasLoadingError] = useState(false);
+
+  const loadUserInfo = async (userID: number) => {
+    setIsLoading(true);
+
+    try {
+      const loadedUser = await getUser(userID);
+
+      setUser(loadedUser);
+    } catch (error) {
+      setHasLoadingError(true);
+    } finally {
+      setIsLoading(false);
+    }
+  };
+
+  useEffect(() => {
+    loadUserInfo(userId);
+  }, []);
```

Comment:
Add userId to the dependency array of the useEffect to avoid potential bugs if userId changes. And move loadUserInfo function inside this useEffect